src/app/services/StudentService.ts

```diff
@@ -28,21 +28,29 @@ export const StudentService = createApi({
         method: "POST",
         body: request,
       }),
+      onQueryStarted: async (_, { dispatch, queryFulfilled }) => {
+        await queryFulfilled;
+        dispatch(UserService.util.invalidateTags(["userStudents"]));
+      },
     }),
     getStudentById: builder.query<StudentDetailResponse, string>({
       query: (id) => `/${id}`,
       providesTags: (_result, _error, id) => [{ type: "Student", id }],
     }),
 
     updateStudent: builder.mutation<void, UpdateStudentRequest>({
-      query: ({ studentId, body }) => ({
+      query: ({ studentId, ...patch }) => ({
```

Comment:
Mas que "patch" lo cambiaria a payload, muy bien aca en el uso de la desestructuracion de datos


```suggestion
      query: ({ studentId, ...payload }) => ({
```